Report errors as GitHub Actions Annotations

Apparently if you run ESLint on GitHub Actions it automatically changes the output and GitHub reports the errors inline:

<img width="662" alt="Screen Shot 2020-04-18 at 12 13 32" src="https://user-images.githubusercontent.com/1402241/79634976-31327f00-816e-11ea-9863-81378c3622d4.png">

On my computer
```
$ eslint .

./packages/helper-settings/components/checkbox.js
  13:7  error  Form label must have ALL of the following types of associated control: nesting, id  jsx-a11y/label-has-for

✖ 1 problem (1 error, 0 warnings)
```

On GHA:

```
$ eslint .

/home/runner/work/OctoLinker/OctoLinker/packages/helper-settings/components/checkbox.js
##[error]  13:7  error  Form label must have ALL of the following types of associated control: nesting, id  jsx-a11y/label-has-for

✖ 1 problem (1 error, 0 warnings)
```

I can't find any references to it in ESLint’s code and repo, but it appears to happen automatically. Real life example: [PR Actions tab](https://github.com/OctoLinker/OctoLinker/pull/895/checks?check_run_id=596965393), [PR Files tab](https://github.com/OctoLinker/OctoLinker/pull/895/files#diff-7ed373a9d8450195e9632d0c184d41fdR13)

Here's [the docs](https://help.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-error-message), but it doesn't seem to match ESLint's output (`##[error]` vs `::error`)

My guess is that GHA is configured to automatically use a different reporter for ESLint, rather than being a feature of ESLint itself.

--- a/cli-main.js
+++ b/cli-main.js
@@ -4,6 +4,7 @@ const updateNotifier = require('update-notifier');
 const getStdin = require('get-stdin');
 const meow = require('meow');
 const formatterPretty = require('eslint-formatter-pretty');
+const formatterCompact = require('eslint/lib/cli-engine/formatters/compact');
 const semver = require('semver');
 const openReport = require('./lib/open-report');
 const xo = require('.');
@@ -136,7 +137,7 @@ if (typeof options.space === 'string') {
 }
 
 const log = report => {
-	const reporter = options.reporter ? xo.getFormatter(options.reporter) : formatterPretty;
+	const reporter = options.reporter ? xo.getFormatter(options.reporter) : process.env.GITHUB_ACTIONS ? formatterCompact : formatterPretty;
 	process.stdout.write(reporter(report.results));
 	process.exitCode = report.errorCount === 0 ? 0 : 1;
 };

--- a/cli-main.js
+++ b/cli-main.js
@@ -4,7 +4,6 @@ const updateNotifier = require('update-notifier');
 const getStdin = require('get-stdin');
 const meow = require('meow');
 const formatterPretty = require('eslint-formatter-pretty');
-const formatterCompact = require('eslint/lib/cli-engine/formatters/compact');
 const semver = require('semver');
 const openReport = require('./lib/open-report');
 const xo = require('.');
@@ -137,7 +136,7 @@ if (typeof options.space === 'string') {
 }
 
 const log = report => {
-	const reporter = options.reporter ? xo.getFormatter(options.reporter) : process.env.GITHUB_ACTIONS ? formatterCompact : formatterPretty;
+	const reporter = options.reporter || process.env.GITHUB_ACTIONS ? xo.getFormatter(options.reporter || 'compact') : formatterPretty;
 	process.stdout.write(reporter(report.results));
 	process.exitCode = report.errorCount === 0 ? 0 : 1;
 };

